src/python/pants/backend/python/goals/setup_py_test.py

```diff
@@ -227,6 +227,129 @@ def test_generate_chroot(chroot_rule_runner: RuleRunner) -> None:
     )
 
 
+def test_generate_chroot_entry_points(chroot_rule_runner: RuleRunner) -> None:
+    chroot_rule_runner.add_to_build_file(
+        "src/python/foo/bar/baz",
+        textwrap.dedent(
+            """
+            python_distribution(
+                name="baz-dist",
+                dependencies=[':baz'],
+                provides=setup_py(
+                    name='baz',
+                    version='1.1.1'
+                )
+            )
+
+            python_library()
+            """
+        ),
+    )
+    chroot_rule_runner.create_file("src/python/foo/bar/baz/baz.py")
+    chroot_rule_runner.add_to_build_file(
+        "src/python/foo/qux",
+        textwrap.dedent(
+            """
+            python_library()
+
+            pex_binary(name="bin", entry_point="foo.qux.bin:main")
+            """
+        ),
+    )
+    chroot_rule_runner.create_file("src/python/foo/qux/__init__.py")
+    chroot_rule_runner.create_file("src/python/foo/qux/qux.py")
+    # Add a `.pyi` stub file to ensure we include it in the final result.
+    chroot_rule_runner.create_file("src/python/foo/qux/qux.pyi")
+    chroot_rule_runner.add_to_build_file(
+        "src/python/foo/resources", 'resources(sources=["js/code.js"])'
+    )
+    chroot_rule_runner.create_file("src/python/foo/resources/js/code.js")
+    chroot_rule_runner.add_to_build_file("files", 'files(sources=["README.txt"])')
+    chroot_rule_runner.create_file("files/README.txt")
+    chroot_rule_runner.add_to_build_file(
+        "src/python/foo",
+        textwrap.dedent(
+            """
+            python_distribution(
+                name='foo-dist',
+                # TODO: remove this once dep inference from entry points works
```

Comment:
Is there a tracking issue you can reference here?

Reply:
Yes, #11807

Reply:
Resolved this in this PR.